pkg/compactv2/modifiers.go

```diff
@@ -62,12 +62,9 @@ SeriesLoop:
 		for _, deletions := range d.d.deletions {
 			for _, m := range deletions.Matchers {
 				v := lbls.Get(m.Name)
-				if v == "" {
-					continue
-				}
 
 				// Only if all matchers in the deletion request are matched can we proceed to deletion.
-				if !m.Matches(v) {
+				if v == "" || !m.Matches(v) {
```

Comment:
`lbls.Get(m.Name)` returns an empty string if the label is not present in the label set. What if the label name is present but with an empty string as its value, is this a valid case in Prometheus?

If yes, we need to use `lbls.Has(m.Name)` here.

Reply:
ack, this is a good point, @yeya24

Reply:
Just checked Prometheus and it is not allowed. So checking `v == ""` is fine.

Reply:
that's right, having empty value is the same as not having the label at all in Prometheus